vng/vector/primitive.go

```diff
@@ -2,37 +2,83 @@ package vector
 
 import (
 	"errors"
+	"fmt"
 	"io"
 
 	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/order"
+	"github.com/brimdata/zed/runtime/expr"
 	"github.com/brimdata/zed/zcode"
 	"golang.org/x/exp/slices"
 )
 
+const MaxDictSize = 256
+
 type PrimitiveWriter struct {
 	typ      zed.Type
 	bytes    zcode.Bytes
 	spiller  *Spiller
 	segments []Segment
+	dict     map[string]uint32
+	cmp      expr.CompareFn
+	min      *zed.Value
+	max      *zed.Value
+	count    uint32
+	nulls    uint32
+	hasNull  int
 }
 
-func NewPrimitiveWriter(typ zed.Type, spiller *Spiller) *PrimitiveWriter {
+func NewPrimitiveWriter(typ zed.Type, spiller *Spiller, useDict bool) *PrimitiveWriter {
+	var dict map[string]uint32
+	if useDict {
+		dict = make(map[string]uint32)
+	}
 	return &PrimitiveWriter{
 		typ:     typ,
 		spiller: spiller,
+		dict:    dict,
+		cmp:     expr.NewValueCompareFn(order.Asc, false),
 	}
 }
 
 func (p *PrimitiveWriter) Write(body zcode.Bytes) error {
+	p.update(body)
 	p.bytes = zcode.Append(p.bytes, body)
-	var err error
-	if len(p.bytes) >= p.spiller.Thresh {
-		err = p.Flush(false)
+	return nil
+}
+
+func (p *PrimitiveWriter) update(body zcode.Bytes) {
+	p.count++
+	if body == nil {
+		p.nulls++
+		p.hasNull = 1
+		return
+	}
+	if body != nil {
+		val := zed.NewValue(p.typ, body)
+		if p.min == nil || p.cmp(val, p.min) < 0 {
+			p.min = val
+		}
+		if p.max == nil || p.cmp(val, p.max) > 0 {
+			p.max = val
+		}
+	}
```

Comment:
Given the previous if statement, the condition here is always true.
```suggestion
	val := zed.NewValue(p.typ, body)
	if p.min == nil || p.cmp(val, p.min) < 0 {
		p.min = val
	}
	if p.max == nil || p.cmp(val, p.max) > 0 {
		p.max = val
	}
```